### DISCUSSION TITLE
Range `sort` and `rsort` do nothing

### DISCUSSION THREAD

**Discussion by devinschulz** [helix-editor/helix#6841] [Q&A] [answered]
### Summary

`:sort` and `rsort` don't appear to work in any interactions I've tried (characters starting with either numbers or letters). As a long-time vim user, I expected it to work the same.

### Reproduction Steps

<https://asciinema.org/a/FrVjmyIXpxmfyYtyxbKKTmeev>

I tried this:

1. select a few lines of text
2. type `:sort` or `:rsort`
3. nothing happens

I expected this to happen:

The lines should be sorted in order

Instead, this happened: 

Nothing happens

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>
2023-04-21T10:47:52.697 helix_vcs [ERROR] failed to obtain current head name for /<path>/test.txt
2023-04-21T10:48:19.532 helix_view::editor [ERROR] editor error: 1 unsaved buffer(s) remaining: ["test.txt"]
</details>

### Platform

MacOS

### Terminal Emulator

Alacritty 0.11.0

### Helix Version

helix 23.03 (3cf03723)

**Comment by pascalkuthe** [comment, marked as answer]
helixes `:sort` command is more general then vims and doesn't operate on lines and instead sorts the selections. For helix multi cursors/selections are a core part of the editing model. You can create on cursors manually with `c` or use a regex with `s` to create a selection at each regex match within the current selection. You can split an existing selection at line boundaries with A-s. Then using `:sort` afterward matches `vims` sort command but you can also sort words for example: `s\w+<ret>:sort<ret>` selects and then sorts all words in the selections.

**Reply by devinschulz** [reply]
Gotcha, that makes sense. Just my limited understanding of how the editor works. Thanks for clarifying 😀

**Reply by jnns** [reply]
I just noticed that `:pipe sort` (or `|sort`) works as one would expect from vim.

**Reply by James-Firth** [reply]
The description I have is `Sort ranges in selection.` which to me implied it would sort the lines within the selection so I used `%` to select the whole file.

Now that I know how it works I think the in-editor description could just use an update and say something like "Sort multiple selections" or something. 

I think sorting selections is way more flexible now that I know, but the description/command definitely confused me a few times.


#### Helix Version

helix 25.01.1 (09b2f6ab)
